docs(pg-adapter): add unique and foreign key constraints to the PG adapter schema

baptiste00001 opened the pull request:
## ☕️ Reasoning

The contraints in the PG adapter schema are non existant while they are defined in all other adapters. I added them to the schema.

## 🧢 Checklist

- [X] Documentation
- [X] Tests
- [X] Ready to be merged

baptiste00001 committed:
Update schema.sql

Add unique and foreign key constraints to the schema

baptiste00001 committed:
Update pg.mdx

Add unique and foreign key contraints to the schema

baptiste00001 committed:
Update schema.sql

user's email may be NULL

baptiste00001 committed:
Update pg.mdx

user's email may be NULL

baptiste00001 committed:
Update pg.mdx

Bug fix

baptiste00001 committed:
Update schema.sql

Bug fix

ndom91 reviewed (approved):
Nice, thanks!

ndom91 commented:
Ah wait, we may need to remove the unique constraint on email. Will that fail if we try to use `null` for multiple users `email`?

baptiste00001 commented:
> Ah wait, we may need to remove the unique constraint on email. Will that fail if we try to use `null` for multiple users `email`?

It works fine with `null` value for multiple users `email` because there is no NULL NOT DISTINCT after the UNIQUE constraint. 
So two `null` values are considered distinct.
I just tested it.

baptiste00001 committed:
Update schema.sql

For some reason cascade delete does not work without explicitly writing the primary key (id) in the foreign key definition.

baptiste00001 committed:
Update pg.mdx

For some reason cascade delete does not work without explicitly writing the primary key (id) in the foreign key definition.

ndom91 commented:
Approved the test run from fork to ensure the pg schema still works. It ran into a few prettier issues. I know one isn't from a file in your PR, but would you mind running prettier over these here? :pray: 

See: https://github.com/nextauthjs/next-auth/actions/runs/12443333045/job/34755721579?pr=12381

baptiste00001 committed:
Remove spaces to pass prettier formatting check

baptiste00001 committed:
Merge branch 'nextauthjs:main' into main

baptiste00001 committed:
Remove extra space to pass prettier formatting check

baptiste00001 commented:
> I approved the test run from fork to ensure the pg schema still works. It ran into a few prettier issues. ~I know one isn't from a file in your PR, but would you mind running prettier over these here? 🙏~
> 
> ~See: https://github.com/nextauthjs/next-auth/actions/runs/12443333045/job/34755721579?pr=12381~
> 
> EDIT: Neverind, fixed it [here](https://github.com/nextauthjs/next-auth/pull/12414)

Oh OK
I actually fixed it as well and commited.
It was a bunch of extra spaces that needed to be removed in two files.
I hope you can merge the PR now.
If you need anything else just ask me.
